### DISCUSSION TITLE
Mapping to open URL/file under the cursor

### DISCUSSION THREAD

**Discussion by basilioss** [zk-org/zk-nvim#132] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

In the README there is a mapping (`map("n", "<CR>", "<Cmd>lua vim.lsp.buf.definition()<CR>", opts)`) to open the link under the caret but it doesn't open URLs and other files like images. Is there an alternative that can open links other than markdown files?

**Comment by mickael-menu** [comment]
This is a bit out of scope for the project. But you can (natively in Neovim) use `gx` to open an URL, and `gf` to open a relative file path (e.g. for an image).

It would be interesting to have some fallbacks with the `definition()` mapping, but I'm not sure how to do that.

**Reply by basilioss** [reply]
`gf` opens binary files in the editor for me. Is it possible to open files with `xdg-open` like vimwiki does?

**Reply by mickael-menu** [reply]
Yeah actually this is what `gx` does, as far as I know, so that could work.

**Reply by basilioss** [reply]
It seems like `gx` doesn't work with relative links. I also use nvim-tree and have netrw (`gx`) disabled.

**Comment by eric-hansen** [comment]
You could read the text under the cursor and see if it's a note reference.
If so then handle accordingly.  Otherwise pass it over to vim's gx handler

**Comment by kha-dinh** [comment]
I'm using [MkdnFlow](https://github.com/jakewvincent/mkdnflow.nvim#follow-links-and-citations) for navigation, which provides a kind of universal follow link for citations, URLs and notes. There is probably many overlapping features with zk.
